tests/token-vesting_test.ts

```diff
@@ -1,180 +1,228 @@
+    this.chain = chain;
+    this.deployer = deployer;
+  }
 
-    balanceOf = (wallet: string) => {
-        const block = this.chain.callReadOnlyFn(vestingOptions.token, "get-balance", [
-            types.principal(wallet),
-        ], this.deployer.address);
+  balanceOf = (wallet: string) => {
+    const block = this.chain.callReadOnlyFn(vestingOptions.token, "get-balance", [
+      types.principal(wallet),
+      ], this.deployer.address);
 
-        return block;
-    }
+    return block;
+  }
 }
 
 Clarinet.test({
-    name: "[deposit] locked amount is transferred to the contract address",
-    async fn(chain: Chain, accounts: Map<string, Account>) {
-        const deployer = accounts.get("deployer")!;
-        const assignees = [{address: accounts.get("wallet_3")!.address, amount: 25}];
-        let assigneesList: any[] = [];
-        assignees.forEach((el) => {
-            assigneesList.push(
-                types.tuple({
-                    'address': types.principal(el.address),
-                    'amount': types.uint(el.amount)
-                })
-            )
-        });
-
-        const block = chain.mineBlock(
-            [
-                Tx.contractCall(
-                    contract, "deposit",
-                    [
-                        types.principal(`${deployer.address}.${vestingOptions.token}`),
-                        types.uint(vestingOptions.amount),
-                        types.uint(vestingOptions.lockingPeriod),
-                        types.list(assigneesList)
-                    ],
-                    deployer.address
-                ),
-            ]
-        );
-
-        let resp = block.receipts[0];
-        resp.result.expectOk().expectBool(true);
-
-        // Check balance in the contract address.
-        const xyzToken = new XyzToken(chain, deployer);
-        resp = xyzToken.balanceOf(`${deployer.address}.${contract}`);
-        resp.result.expectOk().expectUint(vestingOptions.amount);
-    }
+  name: "[deposit] locked amount is transferred to the contract address",
+  async fn(chain: Chain, accounts: Map<string, Account>) {
+    const deployer = accounts.get("deployer")!;
+    const assignees = [{address: accounts.get("wallet_3")!.address, amount: 25}];
+    let assigneesList: any[] = [];
+    assignees.forEach((el) => {
+      assigneesList.push(
+        types.tuple({
+          'address': types.principal(el.address),
+          'amount': types.uint(el.amount)
+        })
+        )
+    });
+
+    let block = chain.mineBlock(
+      [
+        Tx.contractCall(
+          contract, "deposit",
+          [
+            types.principal(`${deployer.address}.${vestingOptions.token}`),
+            types.uint(vestingOptions.amount),
+            types.uint(vestingOptions.lockingPeriod),
+            types.list(assigneesList)
+          ],
+          deployer.address
+        ),
+      ]
+    );
+
+    let resp = block.receipts[0];
+    resp.result.expectErr().expectUint(12);
```

Comment:
Seems better to check the whitelist guard in a separate test case. That will free other tests from the need to check if a token is whitelisted. What you think?